flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ExplainDetailOperation.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.flink.table.operations;
+
+import org.apache.flink.util.StringUtils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** explain operation. */
+public class ExplainDetailOperation implements Operation {
+    private Operation child;
+    private List<String> explainDetails;
+
+    public ExplainDetailOperation(Operation child, List<String> explainDetails) {
+        this.child = child;
+        this.explainDetails = explainDetails;
+    }
+
+    public Operation getChild() {
+        return child;
+    }
+
+    public List<String> getExplainDetails() {
+        return explainDetails;
+    }
+
+    @Override
+    public String asSummaryString() {
+        String detailsString =
```

Comment:
Why use `detailString` as the variable name? 

The name of the variable should represent its conecpt rather than implementaion. Because implementation may change but the concept is stable.